### PR TITLE
feat(AdvanceSteel): Mechanism to get main properties of native objects

### PR DESCRIPTION
**Pull request by phliberato**
## Changes:

- Mechanism(by reflection) to get main properties of native objects
- PropertyInfo and MethodInfo cache
- Copying the right Triangle.dll nuget library to Kit folder

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

**Review by AlanRynne — CHANGES_REQUESTED** [review]
Haven't checked the actual code, but we're doing a repo-wide reformatting and cleanup and it seems some of your settings are conflicting with ours.

For formatting, we're using https://csharpier.com now (which is enforced by pre-commit hooks that are automatically installed).

If you wish to use Rider's Cleanup feature, we have a custom cleanup profile that you should set as the default for the repo (it should be already, but your settings may have overriden that)

File: Objects/Objects/BuiltElements/AdvanceSteel/AsteelBeam.cs [hunk 1]
```diff
@@ -5,31 +5,33 @@
 using Speckle.Core.Kits;
 using Speckle.Core.Models;
 
-namespace Objects.BuiltElements.AdvanceSteel;
-
-public class AsteelBeam : Beam, IDisplayValue<List<Mesh>>, IHasVolume, IHasArea, IAsteelObject
+namespace Objects.BuiltElements.AdvanceSteel
```
[the remark this answers is not in the document]
Comment (phliberato) [reply Objects/Objects/BuiltElements/AdvanceSteel/AsteelBeam.cs:8]: Sorry Alan! When I was merging the branches, I thought it was a conflict error of git…. 😂

[the remark this answers is not in the document]
Comment (phliberato) [reply Objects/Objects/BuiltElements/AdvanceSteel/AsteelBeam.cs:8]: Could you fix this for me?

[the remark this answers is not in the document]
Comment (AlanRynne) [reply Objects/Objects/BuiltElements/AdvanceSteel/AsteelBeam.cs:8]: Yeah sorry about those conflicts, we should have told you about this incoming stuff, but we were still prepping the last details on our side!

I'll see if I can run a fix on this tomorrow

**Comment by teocomi** [comment]
Hey @phliberato, apologies this got a bit forgotten, let us know if it's got to be merged in DEV for the 2.16 release :)

**Comment by phliberato** [comment]
> Hey @phliberato, apologies this got a bit forgotten, let us know if it's got to be merged in DEV for the 2.16 release :)

Yes please.

**Review by teocomi — APPROVED** [review]

**Review by teocomi — APPROVED** [review]

**Pull request merged**